generate.py

```diff
@@ -42,6 +42,10 @@ def generate(
     empty[:T] = idx
     idx = empty
 
+    if idx.device.type == "xla":
```

Comment:
Don't forget about this file! :P

Reply:
I think that even initializing a model will create xla ops, so we should add `xm.mark_step()` back to `chat.py`